sklearn/utils/tests/test_optimize.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+from sklearn.utils.optimize import newton_cg
+from sklearn.utils.testing import assert_almost_equal
+
+from scipy.optimize import fmin_ncg
+
+A = np.random.normal(size=(10, 10))
+
+def func(x):
+    Ax = A.dot(x)
+    return .5*(Ax).dot(Ax)
+
+def grad(x):
+    return A.T.dot(A.dot(x))
+
+def hess(x):
+    return lambda x: A.T.dot(A.dot(x))
+
+def func_grad_hess(x):
+    return func(x), grad(x), hess(x)
+
+x0 = np.ones(10)
+
+assert_almost_equal(newton_cg(func_grad_hess, func, grad, x0), fmin_ncg(f=func, x0=x0, fprime=grad, fhess=hess))
```

Comment:
I think the last should be ``fhessp`` according to the docs:
http://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_ncg.html
``fhess`` is a function returning a matrix, ``fhessp`` is a function returning a matrix.